lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -372,9 +372,17 @@ public function removeElement($element)
                 return $this->collection->removeElement($element);
             }
 
+            if ($this->em->getUnitOfWork()->getEntityState($element) === UnitOfWork::STATE_NEW) {
+                return true;
```

Comment:
Do we have a test covering this scenario? If not, can you provide one as well?

Reply:
Yes, i had to change a lot in `ExtraLazyCollecitonTest.php` as this change introduced many failures, its fully tested after the adjustment of tests.